datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -169,59 +170,26 @@ def __init__(self, instance, init_config, remapper=None, logger=None):
-                else:
-                    if auth_type != 'basic':
-                        self.logger.debug('auth_type %s is not supported, defaulting to basic', auth_type)
-                    auth = (config['username'], config['password'])
+        auth_type = config['auth_type'].lower()
+        if auth_type not in AUTH_TYPES:
+            self.logger.warning('auth_type %s is not supported, defaulting to basic', auth_type)
+            auth_type = 'basic'
 
-            elif config['ntlm_domain']:
-                ensure_ntlm()
-
-                auth = requests_ntlm.HttpNtlmAuth(config['ntlm_domain'], config['password'])
-
-        if auth is None:
+        if auth_type == 'basic':
             if config['kerberos_auth']:
-                ensure_kerberos()
-
-                # For convenience
-                if is_affirmative(config['kerberos_auth']):
-                    config['kerberos_auth'] = 'required'
-
-                if config['kerberos_auth'] not in KERBEROS_STRATEGIES:
-                    raise ConfigurationError(
-                        'Invalid Kerberos strategy `{}`, must be one of: {}'.format(
-                            config['kerberos_auth'], ' | '.join(KERBEROS_STRATEGIES)
-                        )
-                    )
-
-                auth = requests_kerberos.HTTPKerberosAuth(
-                    mutual_authentication=KERBEROS_STRATEGIES[config['kerberos_auth']],
-                    delegate=is_affirmative(config['kerberos_delegate']),
-                    force_preemptive=is_affirmative(config['kerberos_force_initiate']),
-                    hostname_override=config['kerberos_hostname'],
-                    principal=config['kerberos_principal'],
+                self.logger.warning(
+                    'The ability to use Kerberos auth without explicitly setting auth_type to '
+                    '`kerberos` is deprecated and will be removed in Agent 7.20.0'
                 )
-            elif config['auth_type'].lower() == 'aws':
-                ensure_aws()
-
-                if not config['aws_host']:
-                    raise ConfigurationError('AWS auth requires the setting `aws_host`')
-
-                if not config['aws_region']:
-                    raise ConfigurationError('AWS auth requires the setting `aws_region`')
-
-                if not config['aws_service']:
-                    raise ConfigurationError('AWS auth requires the setting `aws_service`')
-
-                auth = requests_aws.BotoAWSRequestsAuth(
-                    aws_host=config['aws_host'], aws_region=config['aws_region'], aws_service=config['aws_service']
+                auth_type = 'kerberos'
+            elif config['ntlm_domain']:
+                self.logger.warning(
+                    'The ability to use NTLM auth without explicitly setting auth_type to '
+                    '`ntlm` is deprecated and will be removed in Agent 7.20.0'
```

Comment:
```suggestion
                    '`ntlm` is deprecated and will be removed in Agent 8.0.0'
```

Same as the other comment.